src/main/java/stobberi/components/TaskList.java

```diff
@@ -0,0 +1,155 @@
+package stobberi.components;
+
+import stobberi.stobberiException.StobberiException;
+import stobberi.task.Deadline;
+import stobberi.task.Event;
+import stobberi.task.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a list of tasks. Provides methods to manage, display, and filter tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> listOfTasks;
+
+    /**
+     * Constructs a new TaskList with an empty list of tasks.
+     */
+    public TaskList() {
+        this.listOfTasks = new ArrayList<>();
+    }
+
+    /**
+     * Marks a task as done by its number in the list.
+     *
+     * @param number The number of the task to mark as done.
+     */
+    public void markTask(int number) {
+        listOfTasks.get(number - 1).setDone();
+        String done = "Nice! I've marked this task as done:\n" +
+                "  ";
+        done += listOfTasks.get(number - 1).toString();
+        Ui.displayForm(done);
+    }
+
+    /**
+     * Marks a task as not done by its number in the list.
+     *
+     * @param number The number of the task to mark as not done.
+     */
+    public void unmarkTask(int number) {
+        listOfTasks.get(number - 1).setNotDone();
+        String done = "OK, I've marked this task as not done yet:\n" +
+                "  ";
+        done += listOfTasks.get(number - 1).toString();
+        Ui.displayForm(done);
+    }
+
+    /**
+     * Displays the list of all tasks.
+     */
+    public void displayList() {
+        String list = "Here are the tasks in your list:";
+        for (int i = 1; i < listOfTasks.size() + 1; i++) {
+            list += "\n" + i + ". " + listOfTasks.get(i - 1);
+        }
+        Ui.displayForm(list);
+    }
+
+    /**
+     * Deletes a task by its number in the list.
+     *
+     * @param number The number of the task to delete.
+     */
+    public void delete(int number) {
+        Task temp = listOfTasks.get(number - 1);
+        listOfTasks.remove(number - 1);
+        String done = "Noted. I've removed this task:\n" +
+                "  " + temp
+                + "\nNow you have " + listOfTasks.size() + " tasks in the list.";
+        Ui.displayForm(done);
+    }
+
+    /**
+     * Displays the last added task.
+     */
+    public void displayLastAdded() {
```

Comment:
maybe you can rename this to displayLastAddedTask to make it a clearer method name?

Reply:
That is true. I have edited it accordingly.